crates/services/tx_status_manager/src/tx_status_stream.rs

```diff
@@ -86,12 +87,47 @@ impl TxUpdateStream {
         self.state = match state {
             State::Empty => match msg {
                 TxStatusMessage::Status(TransactionStatus::Submitted(s)) => {
-                    State::Initial(TransactionStatus::Submitted(s))
+                    State::Submitted(TransactionStatus::Submitted(s))
                 }
+
+                TxStatusMessage::Status(TransactionStatus::PreConfirmationSuccess(s)) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationSuccess(s))
+                }
+                TxStatusMessage::Status(TransactionStatus::PreConfirmationFailure(s)) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationFailure(s))
+                }
+                TxStatusMessage::Status(
+                    TransactionStatus::PreConfirmationSqueezedOut(s),
+                ) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationSqueezedOut(s))
+                }
+
                 TxStatusMessage::Status(s) => State::EarlySuccess(s),
+
                 TxStatusMessage::FailedStatus => State::Failed,
             },
-            State::Initial(s1) => {
+            State::Submitted(s1) => match msg {
+                TxStatusMessage::Status(TransactionStatus::Submitted(s2)) => {
+                    State::Submitted(TransactionStatus::Submitted(s2))
+                }
+
+                TxStatusMessage::Status(TransactionStatus::PreConfirmationSuccess(
+                    s2,
+                )) => State::Preconfirmed(TransactionStatus::PreConfirmationSuccess(s2)),
+                TxStatusMessage::Status(TransactionStatus::PreConfirmationFailure(
+                    s2,
+                )) => State::Preconfirmed(TransactionStatus::PreConfirmationFailure(s2)),
+                TxStatusMessage::Status(
+                    TransactionStatus::PreConfirmationSqueezedOut(s2),
+                ) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationSqueezedOut(s2))
+                }
```

Comment:
```suggestion
```

The same here, transaction doesn't have future statuses.

Reply:
Done